chartpress.py

```diff
@@ -566,20 +567,38 @@ def publish_pages(chart_name, chart_version, chart_repo_github_path, chart_repo_
     this, it is as we would have a --force-publish-chart by default.
     """
 
-    # clone the Helm chart repo and checkout its gh-pages branch,
-    # note the use of cwd (current working directory)
-    checkout_dir = '{}-{}'.format(chart_name, chart_version)
-    check_call(
-        [
-            'git', 'clone', '--no-checkout',
-            git_remote(chart_repo_github_path),
-            checkout_dir,
-        ],
-        # warning: if echoed, this call could reveal the github token
-        echo=True,
-    )
+    # clone/fetch the Helm chart repo and checkout its gh-pages branch, note the
+    # use of cwd (current working directory)
+    checkout_dir = f'{chart_name}-{chart_version}'
+    if not os.path.isdir(checkout_dir):
+        check_call(
+            [
+                'git', 'clone', '--no-checkout',
+                git_remote(chart_repo_github_path),
+                checkout_dir,
+            ],
+            # warning: if echoed, this call could reveal the github token
+            echo=True,
```

Comment:
The comment makes me think that I don't want this call to be echoed, but the code is explicitly set to echo. Which one is correct?

Reply:
Yeah... I don't know. I don't remember why it could risk showing the GitHub token. I can invest time to investigate this once and for all.

Reply:
Investigation complete. Apparently it relates to this.

```python
def git_remote(git_repo):
    """Return the URL for remote git repository.

    Depending on the system setup it returns ssh or https remote.
    """
    # if not matching something/something
    # such as a local directory ".", then
    # simply return this unmodified.
    if not re.match(r'^[^/]+/[^/]+$', git_repo):
        return git_repo

    github_token = os.getenv(GITHUB_TOKEN_KEY)
    if github_token:
        return f'https://{github_token}@github.com/{git_repo}'
    return f'git@github.com:{git_repo}'
```

If GITHUB_TOKEN is set with a CI system as a secret, it will probably be censored, but I can make it censored in chartpress as well, I'll make that happen.

Reply:
I set it to false, it was a security flaw for anyone using GITHUB_TOKEN in a CI system that wouldn't censor it based on being set as a secure environment variable.

I added a note to make it clear what it was about.

Reply:
I ensured it is censored and added a test to make sure the censoring works as well.